auth-svc/controllers/user-controller.js

```diff
@@ -1,22 +1,40 @@
+const UserRepository = require('../service/db-service/UserRepository');
 const userService = require('../service/user-service');
-const { validationResult } = require('express-validator');
+const mailService = require('../service/mail-service');
+const tokenService = require('../service/token-service');
 const ApiError = require('../exeptions/api-error');
+const UserDto = require('../dtos/user-dto');
 
 class UserController {
   async registration(req, res, next) {
     try {
-      const errors = validationResult(req);
-      if (!errors.isEmpty()) {
-        return next(ApiError.BadRequest('Validation error', errors.array()));
+      const { email } = req.body;
+      const userExists = await UserRepository.findByEmail(email);
+
+      if (userExists) {
+        throw ApiError.BadRequest(`User with email ${email} already exists`);
       }
-      const { email, password, name, last_name, role } = req.body;
-      const userData = await userService.registration(
+
+      const registrationPayload =
+        await userService.buildUserRegistrationPayload(req.body);
+      const user = await UserRepository.createUser(registrationPayload);
+
+      console.log(user);
+
+      await mailService.sendActivationMail(
         email,
-        password,
-        name,
-        last_name,
-        role
+        `${process.env.API_URL}/api/activate/${user.activationLink}`
```

Comment:
does the whole url need to be passed here with the env var? could be a bit cleaner by just passing the activationLink and email to this function.

Reply:
also if the email is not sent is that a problem? you are awaiting here but don't wait for a response. if that's the case, do you even need to await if you don't care about the response?

Reply:
Yeah, that's cleaner.
Added full link to mailService


About waiting for email sending, right now it's not required, but in the future, I was thinking about making email confirmation required for every user registration. That way, we will validate that an email is real, so it will be needed in the future. For now, I removed it.